tests/elections/election_types/score/test_cumulative.py

```diff
@@ -0,0 +1,138 @@
+    assert len([c for s in e_random.get_elected() for c in s]) == 2
+
+
+def test_state_list():
+    e = Cumulative(profile_no_tied_cumulative, m=2)
+    assert e.election_states == states
+
+
+def test_get_profile():
+    e = Cumulative(profile_no_tied_cumulative, m=2)
+    assert e.get_profile(0) == profile_no_tied_cumulative
+    assert e.get_profile(1) == profile_no_tied_cumulative_round_1
+
+
+def test_get_step():
+    e = Cumulative(profile_no_tied_cumulative, m=2)
+    assert e.get_step(1) == (profile_no_tied_cumulative_round_1, states[1])
+
+
+def test_get_elected():
+    e = Cumulative(profile_no_tied_cumulative, m=2)
+    assert e.get_elected(0) == tuple()
+    assert e.get_elected(1) == (frozenset({"A"}), frozenset({"B"}))
+
+
+def test_get_eliminated():
+    e = Cumulative(profile_no_tied_cumulative, m=2)
+    assert e.get_eliminated(0) == tuple()
+    assert e.get_eliminated(1) == tuple()
+
+
+def test_get_remaining():
+    e = Cumulative(profile_no_tied_cumulative, m=2)
+    assert e.get_remaining(0) == (frozenset({"A"}), frozenset({"B"}), frozenset({"C"}))
+    assert e.get_remaining(1) == (frozenset({"C"}),)
+
+
+def test_get_ranking():
+    e = Cumulative(profile_no_tied_cumulative, m=2)
+    assert e.get_ranking(0) == (frozenset({"A"}), frozenset({"B"}), frozenset({"C"}))
+    assert e.get_ranking(1) == (frozenset({"A"}), frozenset({"B"}), frozenset({"C"}))
+
+
+def test_get_status_df():
+    e = Cumulative(profile_no_tied_cumulative, m=2)
+
+    df_0 = pd.DataFrame(
+        {"Status": ["Remaining"] * 3, "Round": [0] * 3},
+        index=["A", "B", "C"],
+    )
+    df_1 = pd.DataFrame(
+        {"Status": ["Elected", "Elected", "Remaining"], "Round": [1] * 3},
+        index=["A", "B", "C"],
+    )
+
+    assert e.get_status_df(0).equals(df_0)
+    assert e.get_status_df(1).equals(df_1)
+
+
+def test_errors():
+    with pytest.raises(ValueError):  # m must be non negative
+        Cumulative(profile_no_tied_cumulative, m=0)
+
+    with pytest.raises(ValueError):  # m must be less than num cands
+        Cumulative(profile_no_tied_cumulative, m=4)
+
+    with pytest.raises(ValueError):  # needs tiebreak
+        Cumulative(profile_tied_cumulative, m=2)
+
+
+def test_validate_profile():
+    with pytest.raises(TypeError):  # must be less than limit
+        profile = PreferenceProfile(ballots=[Ballot(scores={"A": 3})])
+        Cumulative(profile, m=2)
+
+    with pytest.raises(TypeError):  # must be less than total budget
+        profile = PreferenceProfile(ballots=[Ballot(scores={"A": 1, "B": 1, "C": 1})])
+        Cumulative(profile, m=2)
+
+    with pytest.raises(TypeError):  # must be non-negative
+        profile = PreferenceProfile(ballots=[Ballot(scores={"A": -3})])
+        Cumulative(profile, m=1)
+
+    with pytest.raises(TypeError):  # must have scores
+        profile = PreferenceProfile(ballots=[Ballot()])
+        Cumulative(profile, m=2)
```

Comment:
We should probably add the error messages on these errors as well to ensure they are raising the *correct* TypeError (this is more of a best-practices thing than anything wrong with the code)